docs/src/content/hardhat-runner/docs/guides/migrating-from-hardhat-waffle.md

```diff
@@ -1,29 +1,111 @@
+We now recommend using [Hardhat Toolbox], a plugin that bundles a curated set of useful packages. These include [Hardhat Chai Matchers] and [Hardhat Network Helpers], which work as an improved replacement for `hardhat-waffle`.
 
-Migrating to these packages only takes a few minutes, as they were designed as a drop-in replacement. If you do so, you'll get more functionality, like support for Solidity custom errors and native `bigint` support, and a more reliable testing experience.
+Migrating to the Toolbox only takes a few minutes. If you do so, you'll get more functionality, like support for Solidity custom errors and native `bigint` support, and a more reliable testing experience. It will also make it easier for you to keep up to date with our recommended setup.
 
-To learn how to start using them, read [this guide](../../../hardhat-chai-matchers/docs/migrate-from-waffle.md).
+## Migrating to Hardhat Toolbox
 
-## Using the Hardhat Toolbox
+Follow these steps to migrate your project to Hardhat Toolbox.
 
-You can get our recommended setup by installing [`@nomicfoundation/hardhat-toolbox`], a single plugin that has everything you need.
+1. First you'll need to remove some packages from your project.
 
-When you use it, you'll be able to:
+   ::::tabsgroup{options="npm 7+,npm 6,yarn"}
 
-- Deploy and interact with your contracts using [ethers.js](https://docs.ethers.io/v5/) and the [`hardhat-ethers`](/hardhat-runner/plugins/nomiclabs-hardhat-ethers) plugin.
-- Test your contracts with [Mocha](https://mochajs.org/), [Chai](https://chaijs.com/) and our own [Hardhat Chai Matchers](/hardhat-chai-matchers) plugin.
-- Interact with Hardhat Network with our [Hardhat Network Helpers](/hardhat-network-helpers).
-- Verify the source code of your contracts with the [hardhat-etherscan](/hardhat-runner/plugins/nomiclabs-hardhat-etherscan) plugin.
-- Get metrics on the gas used by your contracts with the [hardhat-gas-reporter](https://github.com/cgewecke/hardhat-gas-reporter) plugin.
-- Measure your tests coverage with [solidity-coverage](https://github.com/sc-forks/solidity-coverage).
-- And, if you are using TypeScript, get type bindings for your contracts with [Typechain](https://github.com/dethcrypto/TypeChain/).
+   :::tab{value="npm 7+"}
+
+   ```
+   npm uninstall @nomiclabs/hardhat-waffle ethereum-waffle @nomiclabs/hardhat-ethers @nomiclabs/hardhat-etherscan chai ethers hardhat-gas-reporter solidity-coverage @typechain/hardhat typechain @typechain/ethers-v5 @ethersproject/abi @ethersproject/providers
+   ```
+
+   :::
+
+   :::tab{value="npm 6"}
+
+   ```
+   npm uninstall @nomiclabs/hardhat-waffle ethereum-waffle @nomiclabs/hardhat-ethers @nomiclabs/hardhat-etherscan chai ethers hardhat-gas-reporter solidity-coverage @typechain/hardhat typechain @typechain/ethers-v5 @ethersproject/abi @ethersproject/providers
+   ```
+
+   :::
+
+   :::tab{value=yarn}
+
+   ```
+   yarn remove @nomiclabs/hardhat-waffle ethereum-waffle
```

Comment:
For `yarn` we can't do a big remove like we do with npm, because removing a package that is not installed causes an error. Since yarn doesn't install peer dependencies and the next command just adds everything needed, the problem mentioned in the previous comment is not relevant here, so this is fine.